Add page about docker images (#605)
stackabletech/documentation: modules/concepts/pages/container-images.adoc

= Container images
:ubi: https://catalog.redhat.com/software/base-images
:multi-platform-images: https://docs.docker.com/build/building/multi-platform/
:stackable-image-registry: https://repo.stackable.tech/#browse/browse
:stackable-sboms: https://sboms.stackable.tech/

> The core artifacts of the Stackable Data Platform are container images or Docker images.

Reviewer: ```suggestion
The core artifacts of the Stackable Data Platform are container images of operators and products.
```

The operators as well as the deployed Stacklets are all running using container images.

> All images deployed by the SDP are built by Stackable and non-Stackable images - for example official upstream images - are generally not compatible with Stackable operators.

Reviewer 2: Well, actually there are some upstream images we use for CSI stuff: `docker.stackable.tech/k8s/sig-storage/csi-provisioner` and `docker.stackable.tech/k8s/sig-storage/csi-node-driver-registrar`.
I'd leave it to you if this a dirty secret or if we should say "Most of the images deployed ..."

Author: Phew .. from a _conceptual_ point of view I think it's fair to just omit this; I think say "most" is more confusing that just not mentioning this.

Author: But now I changed the intro anyways so I think we can resolve this?

> All images deployed by the SDP are built by Stackable and non-Stackable images - for example official upstream images - are generally not compatible with Stackable operators.

Reviewer: I propose to drop these two sentences completely

Author: https://github.com/stackabletech/documentation/pull/605/commits/9ac1f74cd1bc0b1ad0f9ba37aea694070ab47b6b

== Images overview

Every operator is packaged into its own image and every product is also packaged into its own, seperate image.
Products that require multiple different processes to run, such as a coordinator and a worker, still only run off of one image;
usually these products also only provide a single artifact that is used to run all processes.

Product images are built for xref:operators:supported_versions.adoc[] of products (Not all product versions are supported by all releases).

All images are stored in the {stackable-image-registry}[Stackable image registry].

== Image structure and contents

All our images are built using the {ubi}[Red Hat Universal Base Image (UBI)] minimal as a base image.

> To support xref:ROOT:kubernetes.adoc[OpenShift], it is required to use a UBI base image.

Reviewer 2: AFAIK, cc @razvan 
```suggestion
To be an xref:ROOT:kubernetes.adoc[OpenShift] cerified operator, it is required to use a UBI base image.
```

> To support xref:ROOT:kubernetes.adoc[OpenShift], it is required to use a UBI base image.

Reviewer: ```suggestion
This is a requirement for platform to achieve xref:ROOT:kubernetes.adoc[OpenShift] certification and be available in the RedHat Certified Operator catalog.
```

The base image also contains only minimal dependencies and is vetted by RedHat.

The file system structure is the same across all images, which makes the images easier to work with.

> Products are either built from scratch or official artifacts are used.

Reviewer 2: ```suggestion
Products are either built from source code or official artifacts are used.
```

Beyond that, Stackable also adds plugins or extensions that are not shipped by default, to support features such as xref:operators:monitoring.adoc[] or xref:opa:index.adoc[OpenPolicyAgent] support.

> Since Stackable release 24.3, Stackable provides {stackable-sboms}[SBOMs] for all images.

Reviewer: less repetition
```suggestion
Since Stackable release 24.3, {stackable-sboms}[SBOMs] for all images are provided.
```

Have a look at the xref:guides:viewing-and-verifying-sboms.adoc[] guide to learn how to use SBOMs.

[#multi-platform-support]
== Multi-platform support

Starting with Stackable release 24.7, all images are {multi-platform-images}[multi-platform images], supporting the AMD64 and ARM64 architectures.

[#signatures]
== Image signatures

Image signatures help to ensure the authenticity and integrity of container images.
All Stackable images are signed. 
You can xref:guides:enabling-verification-of-image-signatures.adoc[verify image signatures automatically] in your cluster to make sure that the images you are running are authentic and intact.

== Why not use upstream images?

> Some (but not all) products on the Stackable Data Platform already provide Docker images upstream.

Reviewer: One important aspect is that the Stackable team does detailed security and risk assessments on all images shipped by the SDP.

Author: https://github.com/stackabletech/documentation/pull/605/commits/96530a6d5e28c39d00ff8af929fccd8bcfd673f1
